client/my-sites/site-settings/form-writing.jsx

```diff
@@ -161,39 +165,60 @@ const connectComponent = connect(
 	{ pure: false }
 );
 
-const getFormSettings = partialRight( pick, [
-	'default_post_format',
-	'custom-content-types',
-	'jetpack_testimonial',
-	'jetpack_portfolio',
-	'infinite-scroll',
-	'infinite_scroll',
-	'infinite_scroll_google_analytics',
-	'minileven',
-	'wp_mobile_excerpt',
-	'wp_mobile_featured_images',
-	'wp_mobile_app_promos',
-	'post_by_email_address',
-	'after-the-deadline',
-	'onpublish',
-	'onupdate',
-	'guess_lang',
-	'Bias Language',
-	'Cliches',
-	'Complex Expression',
-	'Diacritical Marks',
-	'Double Negative',
-	'Hidden Verbs',
-	'Jargon Language',
-	'Passive voice',
-	'Phrases to Avoid',
-	'Redundant Expression',
-	'ignored_phrases',
-	'photon',
-	'carousel',
-	'carousel_background_color',
-	'carousel_display_exif'
-] );
+const getFormSettings = settings => {
+	const formSettings = pick( settings, [
+		'default_post_format',
+		'custom-content-types',
+		'jetpack_testimonial',
+		'jetpack_portfolio',
+		'infinite-scroll',
+		'infinite_scroll',
+		'infinite_scroll_google_analytics',
+		'minileven',
+		'wp_mobile_excerpt',
+		'wp_mobile_featured_images',
+		'wp_mobile_app_promos',
+		'post_by_email_address',
+		'after-the-deadline',
+		'onpublish',
+		'onupdate',
+		'guess_lang',
+		'Bias Language',
+		'Cliches',
+		'Complex Expression',
+		'Diacritical Marks',
+		'Double Negative',
+		'Hidden Verbs',
+		'Jargon Language',
+		'Passive voice',
+		'Phrases to Avoid',
+		'Redundant Expression',
+		'ignored_phrases',
+		'photon',
+		'carousel',
+		'carousel_background_color',
+		'carousel_display_exif',
+		'date_format',
+		'start_of_week',
+		'time_format',
+		'timezone_string',
+	] );
+
+	// handling `gmt_offset` and `timezone_string` values
+	const gmt_offset = settings.gmt_offset;
```

Comment:
The same logic is currently being duplicated in the general form and the writing form. May I suggest:
* Taking this logic out in a separate utility method.
* Adding tests for that method